From 8b7d4dc8a8c4f2e8c38f8c4d0d5d7b4ab05c0f65 Mon Sep 17 00:00:00 2001
From: At30c <PabloAtsoc9993@outlook.com>
Date: Wed, 16 Sep 2026 22:35:00 -0300
Subject: [PATCH] Disable loopback event consumer on legacy BPF kernels

The Exynos 990 Linux 4.19 kernel does not expose the loopback-access BPF
ring buffer. Skip its native consumer while leaving the rest of
ConnectivityService initialization intact.

---
 .../connectivity/com/android/server/ConnectivityService.smali | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/smali/android/net/connectivity/com/android/server/ConnectivityService.smali b/smali/android/net/connectivity/com/android/server/ConnectivityService.smali
index 0000000..0000000 100644
--- a/smali/android/net/connectivity/com/android/server/ConnectivityService.smali
+++ b/smali/android/net/connectivity/com/android/server/ConnectivityService.smali
@@ -40862,11 +40862,9 @@
     invoke-virtual {v3}, Landroid/net/connectivity/com/android/server/BpfNetMaps;->isLoopbackAccessMetricsEnabled()Z
 
     move-result v3
 
-    if-eqz v3, :cond_9
+    goto :cond_9
 
-    invoke-static {}, Landroid/net/connectivity/com/android/server/connectivity/BpfEventPoller;->nativeInitLoopbackEventConsumer()V
-
     :cond_9
     iget-object v3, p0, Landroid/net/connectivity/com/android/server/ConnectivityService;->mLocalNetEventListener:Landroid/net/connectivity/com/android/server/connectivity/LocalNetEventListener;
 
-- 
2.43.0
